File: FindContinuousSequence.py
# -*- coding:utf-8 -*-
import math
class Solution:
    def FindContinuousSequence(self, tsum):
        arr = []
        if tsum < 1:
            return []
        for i in range(1,math.ceil(tsum/2)):
            sum = 0
            sub_arr = []
            while sum < tsum:
                sub_arr.append(i)
                sum += i
                i += 1
                if sum == tsum:
                    arr.append(sub_arr)
                    break
        return arr

s = Solution()
print(s.FindContinuousSequence(1))

File: cpp/findContinuousSequence.cpp
class Solution {
public:
    vector<vector<int>> findContinuousSequence(int target) {
        vector<vector<int>> r;
        int sum = 0;
        for(int i=1; i< target/2+1; i++){
            sum = i;
            vector<int> arr = {i};
            int j = i+1;
            while(sum<target){                
                sum += j;
                arr.push_back(j);
                if(sum == target){
                    r.push_back(arr);
                    break;
                }
                j++;
            }
        }
        return r;

    }
};